### PR TITLE
:bug: Fix probable bug: imported package name as name identifier

### PR DESCRIPTION
**Pull request by cupakob**
Should be avoided because the package exported identifiers can't be used after the declaration. This may be also confusing for the reader.

Variable renaming is aligned to the official golang guidelines: 'Variable names in Go should be short rather than long. This is especially true for local variables with limited scope. Prefer c to lineCount. Prefer i to sliceIndex'.
Source: https://go.dev/wiki/CodeReviewComments#variable-names

## Related issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by mjudeikis** [comment]
I know ideomatic go kinda states how this should be done, but in my view when I see `c` `b`, `cl` - its kinda mentally required to go "up" and see what it is. So in my head I am usually willing to compromise idiomatic go in favor of readability. 
Not sure what @embik @sttts  things

**Comment by cupakob** [comment]
@mjudeikis if you have suggestions - let me know, I prefer also a little bit longer names than `c`, `s` and `i`.

**Comment by mjudeikis** [comment]
Will review in more details bit later. I think some suggestions here I agree, some are bit on the personal line. So will come back to you

File: cli/cmd/kubectl-kcp/main.go [hunk 1]
```diff
@@ -29,8 +29,8 @@ func main() {
 	flags := pflag.NewFlagSet("kubectl-kcp", pflag.ExitOnError)
 	pflag.CommandLine = flags
 
-	cmd := cmd.KubectlKcpCommand()
-	if err := cmd.Execute(); err != nil {
+	c := cmd.KubectlKcpCommand()
```
Comment (stevekuznetsov) [inline cli/cmd/kubectl-kcp/main.go:32]: `s/c/command/`

File: cmd/sharded-test-server/main.go [hunk 2]
```diff
@@ -223,14 +223,14 @@ func start(proxyFlags, shardFlags []string, logDirPath, workDirPath string, numb
 	// start shards
 	var shards []*shard.Shard
 	for i := 0; i < numberOfShards; i++ {
-		shard, err := newShard(ctx, i, shardFlags, standaloneVW, servingCA, hostIP.String(), logDirPath, workDirPath, cacheServerConfigPath, clientCA)
```
Comment (stevekuznetsov) [inline cmd/sharded-test-server/main.go:226]: IMO better not to alias the test server package as `shard`.

File: cmd/sharded-test-server/third_party/library-go/crypto/crypto.go [hunk 3]
```diff
@@ -723,10 +723,10 @@ func GetServerCert(certFile, keyFile string, hostnames sets.Set[string]) (*TLSCe
 		return nil, err
 	}
 
-	cert := server.Certs[0]
+	c := server.Certs[0]
```
Comment (stevekuznetsov) [inline cmd/sharded-test-server/third_party/library-go/crypto/crypto.go:726]: `s/c/certificate/`

File: cmd/sharded-test-server/third_party/library-go/crypto/crypto.go [hunk 4]
```diff
@@ -1135,22 +1135,22 @@ func encodeKey(key crypto.PrivateKey) ([]byte, error) {
 }
 
 func writeCertificates(f io.Writer, certs ...*x509.Certificate) error {
-	bytes, err := EncodeCertificates(certs...)
+	b, err := EncodeCertificates(certs...)
```
Comment (stevekuznetsov) [inline cmd/sharded-test-server/third_party/library-go/crypto/crypto.go:1138]: `s/b/encoded/`

File: pkg/admission/namespacelifecycle/admission.go [hunk 5]
```diff
@@ -75,15 +75,15 @@ func newWorkspaceNamespaceLifecycle() (*workspaceNamespaceLifecycle, error) {
 		return nil, err
 	}
 
-	lifecycle, err := lifecycle.NewLifecycle(sets.NewString())
```
Comment (mjudeikis) [inline pkg/admission/namespacelifecycle/admission.go:78]: rever. Not worth the effort.

File: pkg/cache/client/context.go [hunk 6]
```diff
@@ -37,9 +37,9 @@ func WithShardInContext(parent context.Context, shard shard.Name) context.Contex
 // ShardFromContext returns the value of the shard key on the ctx,
 // or an empty Name if there is no shard key.
 func ShardFromContext(ctx context.Context) shard.Name {
-	shard, ok := ctx.Value(shardContextKey).(shard.Name)
+	s, ok := ctx.Value(shardContextKey).(shard.Name)
```
Comment (mjudeikis) [inline pkg/cache/client/context.go:40]: revert. Not worth the effort

File: pkg/informer/informer.go [hunk 7]
```diff
@@ -313,18 +313,18 @@ func (d *GenericDiscoveringDynamicSharedInformerFactory[Informer, Lister, Generi
 
 	klog.Background().V(2).WithValues("gvr", gvr).Info("adding dynamic informer for gvr")
 
-	indexers := cache.Indexers{}
+	idxs := cache.Indexers{}
```
Comment (mjudeikis) [inline pkg/informer/informer.go:316]: revert this one too please.

File: pkg/reconciler/cache/replication/replication_reconcile_unstructured.go [hunk 8]
```diff
@@ -139,11 +139,11 @@ func ensureRemaining(cacheObject *unstructured.Unstructured, localObject *unstru
 }
 
 func toUnstructured(obj interface{}) (*unstructured.Unstructured, error) {
-	unstructured := &unstructured.Unstructured{Object: map[string]interface{}{}}
```
Comment (mjudeikis) [inline pkg/reconciler/cache/replication/replication_reconcile_unstructured.go:142]: this one too

File: pkg/virtual/framework/fixedgvs/apiserver/apiserver.go [hunk 9]
```diff
@@ -118,11 +118,11 @@ func (c completedConfig) New(virtualWorkspaceName string, groupManager discovery
 
 	storage := map[string]restStorage.Storage{}
 	for resource, storageBuilder := range c.ExtraConfig.StorageBuilders {
-		restStorage, err := storageBuilder(c.GenericConfig)
```
Comment (mjudeikis) [inline pkg/virtual/framework/fixedgvs/apiserver/apiserver.go:121]: revert this one please

File: test/e2e/conformance/crd_partial_metadata_test.go [hunk 10]
```diff
@@ -350,9 +350,9 @@ type crdConditionsAdapter struct {
 }
 
 func (ca *crdConditionsAdapter) GetConditions() conditionsv1alpha1.Conditions {
-	conditions := conditionsv1alpha1.Conditions{}
```
Comment (mjudeikis) [inline test/e2e/conformance/crd_partial_metadata_test.go:353]: revert this one please.

File: test/e2e/conformance/cross_logical_cluster_list_test.go [hunk 11]
```diff
@@ -257,13 +257,13 @@ func TestCRDCrossLogicalClusterListPartialObjectMetadata(t *testing.T) {
 	require.Eventually(t, func() bool {
 		informers, _ := informerFactory.Informers()
 
-		informer := informers[sheriffsGVR]
```
Comment (mjudeikis) [inline test/e2e/conformance/cross_logical_cluster_list_test.go:260]: this one too.

**Comment by mjudeikis** [comment]
Where I agree most of these are probable bugs, but most IDEs these days catch these once you type them :) readability over everything else should be priority :)

File: cli/cmd/kubectl-kcp/main.go (hunk 1 above)
Comment (cupakob) [reply cli/cmd/kubectl-kcp/main.go:32]: changed

File: cmd/sharded-test-server/main.go (hunk 2 above)
Comment (cupakob) [reply cmd/sharded-test-server/main.go:226]: changed

File: cmd/sharded-test-server/third_party/library-go/crypto/crypto.go (hunk 4 above)
Comment (cupakob) [reply cmd/sharded-test-server/third_party/library-go/crypto/crypto.go:1138]: changed

File: cmd/sharded-test-server/third_party/library-go/crypto/crypto.go (hunk 3 above)
Comment (cupakob) [reply cmd/sharded-test-server/third_party/library-go/crypto/crypto.go:726]: changed

File: pkg/admission/namespacelifecycle/admission.go (hunk 5 above)
Comment (cupakob) [reply pkg/admission/namespacelifecycle/admission.go:78]: done, but what is about alias `lc` for the imported package?

File: pkg/cache/client/context.go (hunk 6 above)
Comment (cupakob) [reply pkg/cache/client/context.go:40]: done, but what is about alias `cshard` or similar?

File: pkg/informer/informer.go (hunk 7 above)
Comment (cupakob) [reply pkg/informer/informer.go:316]: done

File: pkg/virtual/framework/fixedgvs/apiserver/apiserver.go (hunk 9 above)
Comment (cupakob) [reply pkg/virtual/framework/fixedgvs/apiserver/apiserver.go:121]: what is about `rs` alias for the imported package?

File: test/e2e/conformance/crd_partial_metadata_test.go (hunk 10 above)
Comment (cupakob) [reply test/e2e/conformance/crd_partial_metadata_test.go:353]: done

File: test/e2e/conformance/cross_logical_cluster_list_test.go (hunk 11 above)
Comment (cupakob) [reply test/e2e/conformance/cross_logical_cluster_list_test.go:260]: done

**Comment by cupakob** [comment]
@mjudeikis yes, I agree, readability is very important. But just ignoring possible bugs for better readability is not a great solution. I also think, it should be possible to find alternative names (for variables and package aliases).

File: pkg/virtual/framework/fixedgvs/register.go [hunk 12]
```diff
@@ -51,8 +51,8 @@ func (vw *FixedGroupVersionsVirtualWorkspace) Register(vwName string, rootAPISer
 				StorageBuilders: make(map[string]func(apiGroupAPIServerConfig genericapiserver.CompletedConfig) (restStorage.Storage, error)),
 			},
 		}
-		for resourceName, builder := range restStorageBuilders {
-			cfg.ExtraConfig.StorageBuilders[resourceName] = builder
+		for resourceName, bdr := range restStorageBuilders {
```
Comment (mjudeikis) [inline pkg/virtual/framework/fixedgvs/register.go:54]: `s/bdr/restStorageBuilder` ?

I just got idea to look via k8s code base to match where we can

File: pkg/virtual/framework/fixedgvs/register.go [hunk 13]
```diff
@@ -87,12 +87,12 @@ func (vw *FixedGroupVersionsVirtualWorkspace) Register(vwName string, rootAPISer
 		}
 
 		if groupVersionAPISet.OpenAPIDefinitions != nil {
-			spec, err := builder.BuildOpenAPISpecFromRoutes(restfuladapter.AdaptWebServices(server.GenericAPIServer.Handler.GoRestfulContainer.RegisteredWebServices()), config.GenericConfig.OpenAPIConfig)
+			s, err := builder.BuildOpenAPISpecFromRoutes(restfuladapter.AdaptWebServices(server.GenericAPIServer.Handler.GoRestfulContainer.RegisteredWebServices()), config.GenericConfig.OpenAPIConfig)
```
Comment (mjudeikis) [inline pkg/virtual/framework/fixedgvs/register.go:90]: openAPISpec ?

File: pkg/virtual/initializingworkspaces/builder/build.go [hunk 14]
```diff
@@ -342,18 +342,18 @@ func digestUrl(urlPath, rootPathPrefix string) (
 
 	withoutClustersPrefix := strings.TrimPrefix(realPath, "/clusters/")
 	parts = strings.SplitN(withoutClustersPrefix, "/", 2)
-	path := logicalcluster.NewPath(parts[0])
```
Comment (mjudeikis) [inline pkg/virtual/initializingworkspaces/builder/build.go:345]: logicalclusterPath ?

File: pkg/virtual/initializingworkspaces/builder/forwarding.go [hunk 15]
```diff
@@ -198,7 +198,7 @@ func withUpdateValidation(initializer corev1alpha1.LogicalClusterInitializer) re
 	return registry.StorageWrapperFunc(func(resource schema.GroupResource, storage *registry.StoreFuncs) {
 		delegateUpdater := storage.UpdaterFunc
 		storage.UpdaterFunc = func(ctx context.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc, forceAllowCreate bool, options *metav1.UpdateOptions) (runtime.Object, bool, error) {
-			validation := rest.ValidateObjectUpdateFunc(func(ctx context.Context, obj, old runtime.Object) error {
+			v := rest.ValidateObjectUpdateFunc(func(ctx context.Context, obj, old runtime.Object) error {
```
Comment (mjudeikis) [inline pkg/virtual/initializingworkspaces/builder/forwarding.go:201]: validationFunc ?

File: test/e2e/framework/kcp.go [hunk 16]
```diff
@@ -154,16 +154,16 @@ func SharedKcpServer(t *testing.T) RunningServer {
 		// Use a persistent server
 
 		t.Logf("shared kcp server will target configuration %q", kubeconfig)
-		server, err := newPersistentKCPServer(serverName, kubeconfig, TestConfig.ShardKubeconfig(), filepath.Join(RepositoryDir(), ".kcp"))
+		srv, err := newPersistentKCPServer(serverName, kubeconfig, TestConfig.ShardKubeconfig(), filepath.Join(RepositoryDir(), ".kcp"))
```
Comment (mjudeikis) [inline test/e2e/framework/kcp.go:157]: testServer

File: pkg/informer/informer.go (hunk 7 above)
Comment (mjudeikis) [reply pkg/informer/informer.go]: or make cacheIndexers

File: pkg/virtual/framework/fixedgvs/register.go (hunk 12 above)
Comment (cupakob) [reply pkg/virtual/framework/fixedgvs/register.go:54]: sounds good, done

File: pkg/virtual/framework/fixedgvs/register.go (hunk 13 above)
Comment (cupakob) [reply pkg/virtual/framework/fixedgvs/register.go:90]: done

File: pkg/virtual/initializingworkspaces/builder/build.go (hunk 14 above)
Comment (cupakob) [reply pkg/virtual/initializingworkspaces/builder/build.go:345]: done

File: pkg/virtual/initializingworkspaces/builder/forwarding.go (hunk 15 above)
Comment (cupakob) [reply pkg/virtual/initializingworkspaces/builder/forwarding.go:201]: done

File: test/e2e/framework/kcp.go (hunk 16 above)
Comment (cupakob) [reply test/e2e/framework/kcp.go:157]: done

File: pkg/informer/informer.go (hunk 7 above)
Comment (cupakob) [reply pkg/informer/informer.go]: sounds good, done

**Pull request merged**